File: app/src/main/java/com/edotassi/amazmod/adapters/CommandHistoryAdapter.java
package com.edotassi.amazmod.adapters;

import android.annotation.SuppressLint;
import android.content.Intent;
import android.support.annotation.NonNull;
import android.support.annotation.Nullable;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ArrayAdapter;
import android.widget.TextView;

import com.edotassi.amazmod.R;
import com.edotassi.amazmod.db.model.CommandHistoryEntity;
import com.edotassi.amazmod.support.CommandHistoryBridge;

import org.tinylog.Logger;

import java.util.List;

import butterknife.BindView;
import butterknife.ButterKnife;
import butterknife.OnClick;

public class CommandHistoryAdapter extends ArrayAdapter<CommandHistoryEntity> {

    private CommandHistoryBridge bridge;

    public CommandHistoryAdapter(CommandHistoryBridge bridge, int resource, @NonNull List<CommandHistoryEntity> objects) {
        super(bridge.getBridgeContext(), resource, objects);

        this.bridge = bridge;
    }

    @SuppressLint("ClickableViewAccessibility")
    @NonNull
    @Override
    public View getView(int position, @Nullable View convertView, @NonNull ViewGroup parent) {
        View listItem = convertView;
        if (listItem == null) {
            listItem = LayoutInflater.from(getContext()).inflate(R.layout.row_commands, parent, false);
        }

        final CommandHistoryEntity command = getItem(position);

        ViewHolder viewHolder = new ViewHolder(command);
        ButterKnife.bind(viewHolder, listItem);

        viewHolder.value.setText(command.getCommand());

        final View lItem = listItem;

        viewHolder.value.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                String string = command.getCommand();
                Logger.debug("CommandHistoryActivity onClick string: " + string);
                //edit(command, lItem);
                int resultCode = 0;
                Intent resultIntent = new Intent();
                resultIntent.putExtra("COMMAND", string);
                bridge.setResult(resultCode, resultIntent);
                bridge.finish();
            }
        });

        return listItem;
    }

    public class ViewHolder {

        @BindView(R.id.row_commands_value)
        TextView value;

        private CommandHistoryEntity command;

        protected ViewHolder(CommandHistoryEntity command) {
            this.command = command;
        }

        @OnClick(R.id.row_commands_delete)
        public void onDeleteClick() {
            bridge.deleteCommand(command);
        }
    }
}

File: app/src/main/java/com/edotassi/amazmod/adapters/CommandHistoryAdapter.kt
package com.edotassi.amazmod.adapters

import android.content.Intent
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageButton
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import com.edotassi.amazmod.R
import com.edotassi.amazmod.db.model.CommandHistoryEntity
import com.edotassi.amazmod.ui.CommandHistoryActivity
import org.tinylog.kotlin.Logger

class CommandHistoryAdapter(private val mActivity: CommandHistoryActivity, private val commandList: List<CommandHistoryEntity>) : RecyclerView.Adapter<CommandHistoryAdapter.ViewHolder>() {

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        val view = LayoutInflater.from(mActivity).inflate(R.layout.row_commands, parent, false)
        return ViewHolder(view)
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        val cmd = commandList[position]
        holder.setCommand(cmd)
    }

    override fun getItemCount(): Int {
        return commandList.size
    }

    inner class ViewHolder(view: View) : RecyclerView.ViewHolder(view) {
        private lateinit var command: CommandHistoryEntity

        var value: TextView = view.findViewById(R.id.row_commands_value)
        var deleteButton: ImageButton = view.findViewById(R.id.row_commands_delete)

        init {
            deleteButton.setOnClickListener {
                mActivity.deleteCommand(command)
            }

            value.setOnClickListener {
                Logger.debug("CommandHistoryActivity onClick string: ${command.command}")
                val resultCode = 0
                val resultIntent = Intent()
                resultIntent.putExtra("COMMAND", command.command)
                mActivity.setResult(resultCode, resultIntent)
                mActivity.finish()
            }
        }

        fun setCommand(command: CommandHistoryEntity) {
            this.command = command
            value.text = command.command
        }
    }
}